src/orion/core/io/experiment_builder.py

```diff
@@ -0,0 +1,279 @@
+        The user can provide the name of the existing experiment and only provide the changes to
+        apply on it. Here is an minimal example where we fully initialize a first experiment with a
+        config file and then branch from it with minimal information.
+
+        .. code-block:: bash
+
+            # Initialize root experiment
+            orion init_only --config previous_exeriment.yaml ./userscript -x~'uniform(0, 10)'
+            # Branch a new experiment
+            orion hunt -n previous_experiment ./userscript -x~'uniform(0, 100)'
+
+4. Configuration file
+    This configuration file is meant to overwrite the configuration coming from the database.
+    If this configuration file was interpreted as part of the global configuration, a user could
+    only modify an experiment using command line arguments.
+
+5. Command-line arguments
+    Those are the arguments provided to `orion` for any method (hunt, insert, etc). It includes the
+    argument to `orion` itself as well as the user's script name and its arguments.
+
+"""
+import datetime
+import getpass
+import logging
+import os
+
+import orion
+from orion.core.io import resolve_config
+from orion.core.io.database import Database, DuplicateKeyError
+from orion.core.worker.experiment import Experiment, ExperimentView
+
+
+log = logging.getLogger(__name__)
+
+
+class ExperimentBuilder(object):
+    """Builder for :class:`orion.core.worker.experiment.Experiment`
+    and :class:`orion.core.worker.experiment.ExperimentView`
+
+    .. seealso::
+
+        `orion.core.io.experiment_builder` for more information on the process of building
+        experiments.
+
+        :class:`orion.core.worker.experiment.Experiment`
+        :class:`orion.core.worker.experiment.ExperimentView`
+    """
+
+    # pylint:disable=no-self-use
+    def fetch_default_options(self):
+        """Get dictionary of default options"""
+        return resolve_config.fetch_default_options()
+
+    # pylint:disable=no-self-use
+    def fetch_env_vars(self):
+        """Get dictionary of environment variables specific to Oríon"""
+        return resolve_config.fetch_env_vars()
+
+    def fetch_file_config(self, cmdargs):
+        """Get dictionary of options from configuration file provided in command-line"""
+        return resolve_config.fetch_config(cmdargs)
+
+    def fetch_local_config(self, cmdargs):
+        """Get dictionary of options from all local sources (not from db)
+
+        .. seealso::
+
+            `orion.core.io.experiment_builder` for more information on the hierarchy of
+            configurations.
+        """
+        default_options = self.fetch_default_options()
+        env_vars = self.fetch_env_vars()
+        cmdconfig = self.fetch_file_config(cmdargs)
+
+        return resolve_config.merge_configs(default_options, env_vars, cmdconfig, cmdargs)
+
+    def fetch_db_config(self, cmdargs):
+        """Get dictionary of options from all local sources (not from db)
```

Comment:
Either the name of the function is wrong or this comment is misleading, but when we say `fetch db config` we normally expect it to come from the database, which this comment indicates is not what is happening here.

Reply:
Oops, copy-pasted did not edit from method above.